Feature

geraldgsh opened the pull request:
1. Set up a blank HTML document with the appropriate links to your JavaScript and CSS files.

2. Wrote the functions for project creation

3. Wrote for Task creation and project association.

4. Created feature to check, edit and delete individual task

5. Built remaining days timer

6. Installed and configured ESlint

7. Installed and configured Webpack

8. Linted JS and validated HTML & CSS

MauricioRobayo commented on .gitignore:
```diff
@@ -0,0 +1,11 @@
+
+# Lock files
+/package-lock.json
+/yarn.lock
```
You should commit these files: https://stackoverflow.com/questions/44206782/do-i-commit-the-package-lock-json-file-created-by-npm-5

geraldgsh replied:
Noted. I've removed it from gitignore.

MauricioRobayo commented on src/control/projectControl.js:
```diff
@@ -0,0 +1,34 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable no-undef */
+/* eslint-disable no-shadow */
+/* eslint-disable no-empty */
```
You should really try to do your best to adopt the style provided, disabling so many rules on each file is kind of cheating 😅.

geraldgsh replied:
Noted. I've edited the code to satisfy eslint.

MauricioRobayo commented on src/control/projectControl.js:
```diff
@@ -0,0 +1,34 @@
+/* eslint-disable linebreak-style */
+/* eslint-disable no-undef */
+/* eslint-disable no-shadow */
```
This is disabled on the original `.eslintrc.json` that you are supposed to use, so no need to disable it again here: https://github.com/microverseinc/linters-config/blob/master/javascript/.eslintrc.json

geraldgsh replied:
I've added "linebreak-style": "off" rule since it wasn't in the file. Thanks for point it out.

MauricioRobayo commented on src/control/taskControl.js:
```diff
@@ -0,0 +1,112 @@
+/* eslint-disable no-empty */
+/* eslint-disable no-restricted-syntax */
+/* eslint-disable no-undef */
+/* eslint-disable no-shadow */
```
😔.

geraldgsh replied:
I've added "linebreak-style": "off" rule since it wasn't in the file. Thanks for point it out.

MauricioRobayo commented on src/index.js:
```diff
@@ -0,0 +1,105 @@
+/* eslint-disable func-names */
+/* eslint-disable space-before-function-paren */
+/* eslint-disable prefer-arrow-callback */
+/* eslint-disable no-unused-vars */
```
☹.

MauricioRobayo commented on src/modules/bulmaModal.js:
```diff
@@ -0,0 +1,49 @@
+/* eslint-disable func-names */
+/* eslint-disable camelcase */
+/* eslint-disable prefer-arrow-callback */
+/* eslint-disable import/prefer-default-export */
```
It's OK to disable some rule for some special cases, using `// eslint-disable-next-line` or `// eslint-disable-line`.

But disabling so many rules for the whole file doesn't seem appropriate. You're basically skipping the rules provided.

geraldgsh replied:
I've added "linebreak-style": "off" rule since it wasn't in the file. Thanks for point it out.

Please see summarized finds at the bottom of the page.

MauricioRobayo commented on src/modules/bulmaModal.js:
```diff
@@ -0,0 +1,49 @@
+export class BulmaModal {
+  constructor(selector) {
+    this.elem = document.querySelector(selector);
+    this.close_data();
```
The convention for JS variables would be `closeData`. You can rename your functions to follow that convention.

MauricioRobayo commented on src/modules/bulmaModal.js:
```diff
@@ -0,0 +1,49 @@
+    this.on_close();
+  }
+
+  close_data() {
```
`closeData`

geraldgsh replied:
Corrected variables with camelCases

geraldgsh commented:
Thanks @MauricioRobayo for the review. I'll be mindful of items above for other projects.

./src/modules/bulmaModal.js

1. Remove "/* eslint-disable prefer-arrow-callback */"
   Quick fix with arrows
   Modals does not open
   
2.  Remove "/* eslint-disable import/prefer-default-export */"
    export default class BulmaModal
    error: "BulmaModal is not a constructor"

./src/DOM/projectDOM.js

3.  Remove "/* eslint-disable import/prefer-default-export */"
    Added 'module.exports = { addProjectToTaskDropdown, default: addProjectToTaskDropdown };'
    error: "exports" is read-only"

./src/DOM/taskDOM.js

4.  Remove "/* eslint-disable import/prefer-default-export */"
    Added 'module.exports = { renderTaskCard, default: renderTaskCard };'
    error: "exports" is read-only"

./src/index.js

5. Folder contains event listens wrapped in modules that will not be called. Hence, '/* eslint-disable no-unused-vars */' is not ideal.

maelfosso commented on src/control/taskControl.js:
```diff
@@ -0,0 +1,96 @@
+export const iterateTasks = () => {
+  const list = document.getElementById('list');
+  list.innerHTML = '';
+  const projectID = document.getElementById('listProject').value;
```
By supposing that file is the controller, please no DOM access/modification here

maelfosso commented on src/control/taskControl.js:
```diff
@@ -0,0 +1,96 @@
+  const titleInput = document.getElementById('titleInput').value;
+  const dateInput = document.getElementById('dateInput').value;
+  const descInput = document.getElementById('descInput').value;
+  const priorityInput = document.getElementById('priorityInput').value;
```
No DOM instruction here

maelfosso commented on src/control/projectControl.js:
```diff
@@ -0,0 +1,27 @@
+  const projectModal = new BulmaModal('#projectModal');
+  const projectInput = document.getElementById('projectInput').value;
+  if (projectList.some((project) => project.name === projectInput)) {
+    // eslint-disable-next-line no-undef
```
Why this?

geraldgsh commented:
Thanks reviewing my repo @maelfosso

geraldgsh commented:
Thanks @bellom
